components/malloc_size_of/lib.rs

```diff
@@ -782,3 +791,91 @@ impl MallocSizeOf for xml5ever::QualName {
         self.local.size_of(ops)
     }
 }
+
+impl MallocSizeOf for hyper::header::Headers {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let mut t = 0;
+        for x in self.iter() {
+           t += x.size_of(ops)
+        }
+        t
+    }
+}
+
+impl<'a> MallocSizeOf for hyper::header::HeaderView<'a> {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // Close enough?
+        self.name().size_of(ops) + self.value_string().size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::header::ContentType {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // size_of::<hyper::header::ContentType>
+        self.0.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Mime {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.0.size_of(ops) +
+        self.1.size_of(ops) +
+        self.2.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Attr {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        self.len()
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Value {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        self.len() // Length of string value in bytes (not the char length of a string)!
+    }
+}
+
+/*
+ *  pub struct Duration {
+ *      secs: i64,
+ *      nanos: i32,
+ *  }
+ */
+impl MallocSizeOf for time::Duration {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) ->  usize {
+        size_of::<time::Duration>()
+    }
+}
+
+impl MallocSizeOf for time::Tm {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        /*
+        self.tm_sec.size_of(ops) +
+        self.tm_min.size_of(ops) +
+        self.tm_hour.size_of(ops) +
+        self.tm_mday.size_of(ops) +
+        self.tm_mon.size_of(ops) +
+        self.tm_year.size_of(ops) +
+        self.tm_wday.size_of(ops) +
+        self.tm_yday.size_of(ops) +
+        self.tm_isdst.size_of(ops) +
+        self.tm_utcoff.size_of(ops) +
+        self.tm_nsec.size_of(ops)
+        */
+
+        size_of::<time::Tm>()
+    }
+}
+
+impl MallocSizeOf for hyper_serde::Serde<hyper::header::ContentType> {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        (*self).clone().into_inner().size_of(ops)
+    }
+}
+
+impl<T> MallocSizeOf for std::sync::mpsc::Sender<T> {
```

Comment:
This isn't correct, unfortunately. MallocSizeOf for T is supposed to calculate the heap size of the members of T, not the size of T itself. Since we don't know anything about the internals of mpsc::Sender, let's remove this implementation and use `#[ignore_malloc_size_of = "Can't measure mpsc::Sender"]` instead.